### ISSUE TITLE
Has anyone successfully gotten BCM94352Z to work with this model?

### ISSUE THREAD

**Issue by applesjgtl** [syscl/XPS9350-macOS#12]
Bluetooth works with PCIID kext but everything I try to enable wifi (rehabman/thedarkvoid)'s PCIID, various other patches do not. Supposedly latest FakePCIID_Broadcom_WiFi.kext should work without PCI Error patches in Clover config, but card doesn't initialize. See error in DSDT build fail thread screenshot.

**Comment by syscl** [comment]
I know what you lack, you need a patch I will later upload, but now you can added it yourself
```
Kext:    AirPortBrcm4360
Find:    81F952AA00007529 
Replace: 81F952AA00006690
```
Through Clover config.plist:
```

<dict>
    <key>Comment</key>
    <string>AirPortBrcm4360 - fcvo</string>
    <key>Disabled</key>
    <false/>
    <key>Find</key>
    <data>gflSqgAAdSk=</data>
    <key>Name</key>
    <string>AirPortBrcm4360</string>
    <key>Replace</key>
    <data>gflSqgAAZpA=</data>
</dict>
```

syscl

**Comment by syscl** [comment]
Don't forget to use latest FakePCIID from my github

syscl

**Comment by applesjgtl** [comment]
I've tried this patch before unsuccessfully but am trying again now. Also synced with repo using deploy script, pretty sure I was already on latest (mid November) PCIID direct from RehabMan's BitBucket.

**Comment by applesjgtl** [comment]
As expected, no change -- however if I don't remove your audio patches my audio breaks completely. There must be some incompatibility between your solution and the folder attached here https://www.tonymacx86.com/threads/guide-dell-xps-13-9350-macos-10-12-1.204730/ .

**Comment by zeusk** [comment]
I have the DW1560 (BCM94352Z) working; Use the suggested patch in config.plist (At-least it was working last week, haven't tested latest changes but nothing seems to have changed related to WiFi).

**Comment by zeusk** [comment]
So I tried the latest CLOVER and deploy.sh; WiFi works, but bluetooth doesn't. I can turn bluetooth on/off and I did see a device but it never paired and then after a reboot wouldn't even appear in the list but my iPhone connects on the first attempt.

**Comment by syscl** [comment]
@zeusk one second, I now update the bluetooth patch

syscl

**Comment by zeusk** [comment]
@syscl, thanks! will keep an eye out for that.

Also - at-least in the latest version, deploy.sh hangs at updating kexts, I removed _update_clover function from the script and that seemed to have fixed the problem (getting stuck) and manually copied the CLOVER folder to /EFI/

**Comment by syscl** [comment]
@zeusk try latest deploy.sh

**Comment by zeusk** [comment]
@syscl, Thanks! I was running with -v instead of -d (!!)

anyway, it gets stuck asking for user input:

> override rwxrwxrwx  shantanu/staff hidden for /Volumes/EFI/EFI/CLOVER/kexts/10.12/BrcmPatchRAM2.kext/Contents/._Info.plist?

When not in debug mode, this prompt is hidden and it appears as if the script is stuck.

**Comment by zeusk** [comment]
@syscl, also bluetooth now working!

**Comment by syscl** [comment]
@zeusk try the latest deploy.sh

BTW, if you encounter bt issue, just boot to windows:
- turn off
- turn on

Reboot to macOS

syscl

**Comment by zeusk** [comment]
Bluetooth worked, didn't need windows.

**Comment by syscl** [comment]
@zeusk Which commit makes bt work?

syscl

**Comment by zeusk** [comment]
8344f5a worked for me, haven't tried the latest one because I'm using bluetooth speakers now! :D

**Comment by syscl** [comment]
@zeusk I am happy, will roll back to this commit, since ```rm -rf``` is dangerous.

Don't hesitate to report issues and questions, I want to make this project better than ever!

syscl

**Comment by applesjgtl** [comment]
Latest commit seems to have broken bluetooth and sound for me. Going to try manually copying over CLOVER folder but despite DSDT compiling properly nothing seems to have changed.

**Comment by applesjgtl** [comment]
![1481873549782796432948](https://cloud.githubusercontent.com/assets/2176220/21255129/c60d9f04-c31e-11e6-9da5-960a8cedc75d.jpg)
Still not working.

**Comment by syscl** [comment]
@applesjgtl what specs you have?

**Comment by applesjgtl** [comment]
I believe I posted this in audio thread. This project is for Skylake early 2016 right? QHD+, touchscreen, i5, 256GB/8GB On Thu, Dec 15, 2016 at 11:50pm, Zhou Yating < notifications@github.com [notifications@github.com] > wrote:
@applesjgtl [https://github.com/applesjgtl] what specs you have?

**Comment by syscl** [comment]
@applesjgtl I have too many posts to reply, impossible to memorize one of it. You have QHD+ with i5, thus it's pretty sure you have to switch the ig-platform-id with patched CoreDisplay.

syscl

**Comment by gctseng** [comment]
@applesjgtl Did you ever get the Bluetooth to work? I have DW1830, BT doesn't pair and only saw my headphones once, but doesn't see my iPhone. Will also randomly disappear after sleep (as in the system doesn't see the device) even though upon reboot it will be there (just doesn't pair or detect devices).  WiFi works.

**Comment by applesjgtl** [comment]
Bluetooth works fine, always has. WiFi is the problem - even after
switching with DW1830, WiFi isn't detected.

**Comment by zeusk** [comment]
I have i7-6560U QHD+ with DW1560, both bluetooth and WiFi work.

I did have some dropped connections using the latest update but not sure if that is because of an issue on my side/my router or this repo.

**Comment by gctseng** [comment]
@applesjgtl I'm did a complete reinstall of Windows 10 and then will install Sierra again.  I did notice when I installed Windows the BT adapter was not detected (it wasn't even listed under Device Manager).  I went into the bios, disabled BT and then enabled it again (without clicking "Apply", just checked and unchecked the box for BT, and then clicked Apply), the BT adapter then was detected under Windows.  I note I had updated a bunch of drivers and then rebooted, but I didn't see a specific driver being updated for the BT adapter and the WiFi adapter worked as soon as Windows was installed.

Maybe you can try to enable/disable WiFi in the bios as well?

**Comment by applesjgtl** [comment]
@gctseng not a bad idea. Just received another XPS 13 in the mail that came with a bcm94350za so I'll have to give that one a try as well. Also, I think I'm on BIOS 1.44 not 1.4.10 so perhaps updating will make a difference.

**Comment by applesjgtl** [comment]
So I think having the second computer made me figure out why I was having the aforementioned issue - the CLOVER folder attached to the maxecloo post on tonymac contains a DSDT file for a different wifi card that I failed to remove. Because your CLOVER folder wouldn't boot the laptop, I had a hybrid of your patches and that other CLOVER folder. I now suspect an M.2 vs NVME incompatibility and have opened a new issue to discuss this. I was wrong, by the way - both laptops are on 1.4.10 BIOS version. Although I have wiped clean the older laptop, I'm already seeing improvement despite the inability to find the target install drive. I'm actually seeing the airport driver load in the installer without errors and the menu appears, though I can't turn on wifi at this point because the patching is obviously incomplete at the installation stage.
